subsys/bluetooth/host/gatt.c

```diff
@@ -5494,26 +5499,20 @@ int bt_gatt_subscribe(struct bt_conn *conn,
 		}
 
 		/* Check if another subscription exists */
-		if (tmp->value_handle == params->value_handle &&
-		    tmp->value >= params->value) {
-			has_subscription = true;
+		if (tmp->value_handle == params->value_handle) {
+			params->value |= tmp->value;
```

Comment:
What is the intended function of this code?

It just does not seem sensible to me to modify the subscription that is being added based on previous subscriptions. Why should we do it? The code comment "Check if another subscription exists" does not explain what happens here.